### ISSUE TITLE
`_ignored` is not populated for a "scaled_float" field defined as "ignore_malformed"

### ISSUE THREAD

**Issue by cnliao** [elastic/elasticsearch#92594]
### Elasticsearch Version

8.5.3

### Java Version

_bundled_

### OS Version

5.4.0-124-generic

### Problem Description

The `_ignored` field is not populated correctly under the following condition:
- A `scaled_float` field is defined as `"ignore_malformed": true`;
- A document with a malformed value for that field is indexed.

### Steps to Reproduce

```
PUT my-index-000001
{
  "settings": {
    "index.mapping.ignore_malformed": true 
  },
  "mappings": {
    "properties": {
      "a_float": {
        "type": "scaled_float",
        "scaling_factor": 100, 
        "ignore_malformed": true
      },
      "a_long": {
        "type": "long",
        "ignore_malformed": true
      }
    }
  }
}

POST my-index-000001/_doc/1?refresh
{
  "a_float": "hello",
  "a_long": "hello"
}

GET my-index-000001/_search
{}

// Response
{
  "took": 0,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 1,
      "relation": "eq"
    },
    "max_score": 1,
    "hits": [
      {
        "_index": "my-index-000001",
        "_id": "1",
        "_score": 1,
        "_ignored": [
          "a_long"
          // should include "a_float" here
        ],
        "_source": {
          "a_float": "hello",
          "a_long": "hello"
        }
      }
    ]
  }
}
```

**Comment by devnamrits** [comment]
Hi @javanna Can I pick this up?

Also, please usher a little bit regarding this issue.

**Comment by cbuescher** [comment]
@cnliao I just tried your reproduction on 8.6.2 and got the expected result in the `_ignore` field::

```
"_ignored": [
          "a_float",
          "a_long"
        ],
```

Can you check again with 8.6, might be already fixed.

**Comment by cbuescher** [comment]
This was fixed with https://github.com/elastic/elasticsearch/pull/90565 in 8.6, therefore I'm closing this issue.

**Comment by cnliao** [comment]
> GET my-index-000001/_search
> {}

Fix confirmed at 8.6.2. Thank you all!
